### ISSUE TITLE
Formatting no longer works

### ISSUE THREAD

**Issue by bvdmitri** [latex-lsp/texlab#871]
Not sure what happened, I didn't change my configuration or anything, but after my vacation I returned to my document and I cannot longer format it. ~~I tried `texlab@5.3.0` but it does now work either.~~ `texlab@5.3.0` works as expected.

My config:
```lua
------------------- Latex LS settings -----------------
-- requires brew install --cask mactex & the Skim application
-- Read the current directory's path
local pwdhandle = io.popen("pwd")
vim.b.pwd = pwdhandle:read()
pwdhandle:close()

-- Get the basename of the current directory's path
local wdhandle = io.popen(string.format("basename %s", vim.b.pwd))
vim.b.wd = wdhandle:read()
wdhandle:close()

vim.b.texlabwd = string.format("/tmp/texlab-latex/%s", vim.b.wd) 
vim.b.texlab_latexindent_config = string.format("%s/latexindent.yaml", vim.b.pwd)

-- Create texlab working directory if it does not exist
os.execute(string.format("mkdir -p %s", vim.b.texlabwd))

lspconfig.texlab.setup {
  on_attach = on_attach,
  log_level = vim.lsp.protocol.MessageType.Log,
  settings = {
    texlab = {
      auxDirectory = vim.b.texlabwd,
      build = {
        executable = "latexmk",
        args = { "-pdf", "-interaction=nonstopmode", "-synctex=1", string.format("-outdir=%s", vim.b.texlabwd), "%f" },
        onSave = true,
      },
      forwardSearch = {
        executable = "/Applications/Skim.app/Contents/SharedSupport/displayline",
        args = { "%l", "%p", "%f" }
      },
      latexindent = {
        ['local'] = vim.b.texlab_latexindent_config,
        modifyLineBreaks = true
      }
    }
  },
  capabilities = autocomplete
}

```

Debug log level shows that Neovim sends the formatting command to the client but nothing happens. It was working about two weeks ago.

```
[DEBUG][2023-03-23 13:48:33] .../lua/vim/lsp.lua:1381	"LSP[texlab]"	"client.request"	1	"textDocument/formatting"	{  options = {    insertSpaces = true,    tabSize = 2  },  textDocument = {    uri = "file:///Users/bvdmitri/Projects/Latex/phdthesis/contents/02-variational-inference/01-introduction.tex"  }}	<function 1>	42
[DEBUG][2023-03-23 13:48:33] .../vim/lsp/rpc.lua:285	"rpc.send"	{  id = 13,  jsonrpc = "2.0",  method = "textDocument/formatting",  params = {    options = {      insertSpaces = true,      tabSize = 2    },    textDocument = {      uri = "file:///Users/bvdmitri/Projects/Latex/phdthesis/contents/02-variational-inference/01-introduction.tex"    }  }}
[DEBUG][2023-03-23 13:48:33] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  id = 13,  jsonrpc = "2.0"}
[DEBUG][2023-03-23 13:49:43] .../lua/vim/lsp.lua:1381	"LSP[texlab]"	"client.request"	1	"textDocument/formatting"	{  options = {    insertSpaces = true,    tabSize = 2  },  textDocument = {    uri = "file:///Users/bvdmitri/Projects/Latex/phdthesis/contents/02-variational-inference/01-introduction.tex"  }}	<function 1>	42
[DEBUG][2023-03-23 13:49:43] .../vim/lsp/rpc.lua:285	"rpc.send"	{  id = 14,  jsonrpc = "2.0",  method = "textDocument/formatting",  params = {    options = {      insertSpaces = true,      tabSize = 2    },    textDocument = {      uri = "file:///Users/bvdmitri/Projects/Latex/phdthesis/contents/02-variational-inference/01-introduction.tex"    }  }}
[DEBUG][2023-03-23 13:49:43] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  id = 14,  jsonrpc = "2.0"}
```

**Comment by pfoerster** [comment]
@bvdmitri Can you try creating a log file using `texlab -vvvv --log-file=/tmp/texlab.log`, please? It would be helpful to know if `texlab` actually calls `latexindent` or if there is some other problem. There should be a line like `Running latexindent in folder <...> with args: <...>` in the log.

**Comment by maikol-solis** [comment]
I have the same issue with `lsp-mode` in emacs 

```
[Trace - 09:51:04 AM] Sending request 'textDocument/formatting - (29)'.
Params: {
  "textDocument": {
    "uri": "path/to/myfile.tex"
  },
  "options": {
    "tabSize": 4,
    "insertSpaces": true,
    "trimTrailingWhitespace": true,
    "insertFinalNewline": true,
    "trimFinalNewlines": true
  }
}


[Trace - 09:51:04 AM] Received response 'textDocument/formatting - (29)' in 0ms.
Result: null
```

Weirdly, the server does not have any capabilities activated

![image](https://user-images.githubusercontent.com/4078489/227728466-d5df9a74-c8ea-4c0d-9135-de02bc1c894e.png)


```
texlab --version
texlab 5.4.0
```

**Comment by bvdmitri** [comment]
@pfoerster Sure, I can try to give more info and create a log file, but I couldn't understand (from the documentation available) how to pass extra arguments to `texlab` executable while running from the Neovim's LSP mode.

EDIT: Ok, I figured it out. I used the `cmd` field in the LSP configuration to change the default arguments. I get the following output for the `format` command:

```
DEBUG - < {"params":{"options":{"tabSize":2,"insertSpaces":true},"textDocument":{"uri":"file:\/\/\/Users\/bvdmitri\/Projects\/Latex\/phdthesis\/contents\/02-variational-inference\/01-introduction.tex"}},"method":"textDocument\/formatting","id":8,"jsonrpc":"2.0"}
DEBUG - > {"jsonrpc":"2.0","id":8,"result":null}
```
I can manually call the `latexindent` command with the same arguments and it works just fine. The problem seems indeed to be somewhere between `texlab` and `neovim` LSP interaction. Perhaps, it stopped working after I ran `brew upgrade` and `:PackerSync`, but I didn't change my configuration.


Extra outputs for the `build` command just in case:
```
DEBUG - < {"params":{"textDocument":{"uri":"file:\/\/\/Users\/bvdmitri\/Projects\/Latex\/phdthesis\/contents\/02-variational-inference\/01-introduction.tex"}},"method":"textDocument\/build","id":6,"jsonrpc":"2.0"}
DEBUG - > {"jsonrpc":"2.0","id":5,"method":"window/workDoneProgress/create","params":{"token":2}}
DEBUG - < {"result":null,"id":5,"jsonrpc":"2.0"}
DEBUG - > {"jsonrpc":"2.0","method":"$/progress","params":{"token":2,"value":{"cancellable":false,"kind":"begin","message":"file:///Users/bvdmitri/Projects/Latex/phdthesis/main.tex","title":"Building"}}}
DEBUG - > {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Rc files read:","type":4}}
DEBUG - > {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"  NONE","type":4}}
DEBUG - > {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Latexmk: This is Latexmk, John Collins, 17 Mar. 2022. Version 4.77, version: 4.77.","type":4}}
DEBUG - > {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Latexmk: Nothing to do for '/Users/bvdmitri/Projects/Latex/phdthesis/main.tex'.","type":4}}
DEBUG - > {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Latexmk: All targets (/private/tmp/texlab-latex/phdthesis/main.pdf) are up-to-date","type":4}}
DEBUG - > {"jsonrpc":"2.0","method":"$/progress","params":{"token":2,"value":{"kind":"end"}}}
DEBUG - > {"jsonrpc":"2.0","id":6,"result":{"status":0}}
```
and for the `search` command:
```
DEBUG - < {"params":{"position":{"character":1,"line":0},"textDocument":{"uri":"file:\/\/\/Users\/bvdmitri\/Projects\/Latex\/phdthesis\/contents\/02-variational-inference\/01-introduction.tex"}},"method":"textDocument\/forwardSearch","id":7,"jsonrpc":"2.0"}
DEBUG - Executing forward search: /Applications/Skim.app/Contents/SharedSupport/displayline ["1", "/tmp/texlab-latex/phdthesis/main.pdf", "/Users/bvdmitri/Projects/Latex/phdthesis/contents/02-variational-inference/01-introduction.tex"]
DEBUG - > {"jsonrpc":"2.0","id":7,"result":{"status":0}}
```

**Comment by bvdmitri** [comment]
I do **not** get the `Running latexindent in folder <...> with args: <...>` line in my logs.

**Comment by bvdmitri** [comment]
@pfoerster I did try again different version from the [releases](https://github.com/latex-lsp/texlab/releases) page and found out that `5.3.0` version works as expected and only `5.4.0` does not respond to the `formatting` request properly. I tried `5.3.0` before opennng the issue and thought it didn't work, but I must have not configured the environment properly. 

So: 
- `5.3.0` formatting works
- `5.4.0` formatting is broken
 
For `5.3.0` I also get the `Running latexindent in folder <...> with args: <...>` line in my logs.
